local-cluster/tests/local_cluster_slow_1.rs

```diff
@@ -386,10 +391,360 @@ fn test_kill_partition_switch_threshold_progress() {
+            }
+        }
+        sleep(Duration::from_millis(1000));
+    }
+    disable_turbine.store(false, Ordering::Relaxed);
+
+    // Send the node the other version of the shred so they realize it's duplicate
+    info!("Resending duplicate shreds to duplicate fork validator");
+    let send_socket = UdpSocket::bind("0.0.0.0:0").unwrap();
+    let duplicate_fork_validator_tvu = cluster.get_contact_info(pubkey).unwrap().tvu;
+    send_socket
+        .send_to(dup_shred1.payload().as_ref(), &duplicate_fork_validator_tvu)
+        .unwrap();
+    send_socket
+        .send_to(dup_shred2.payload().as_ref(), &duplicate_fork_validator_tvu)
+        .unwrap();
+
+    // Check the validator detected a duplicate proof
+    info!("Waiting on duplicate fork validator to see duplicate shreds and make a proof",);
+    loop {
+        let duplicate_fork_validator_blockstore = open_blockstore(&ledger_path);
+        if let Some(dup_proof) = duplicate_fork_validator_blockstore.get_first_duplicate_proof() {
+            assert_eq!(dup_proof.0, dup_slot);
+            break;
+        }
+        sleep(Duration::from_millis(1000));
+    }
+}
+
+
+// We want to simulate the following:
+//   /--- 1 --- 3 (duplicate block)
+// 0
+//   \--- 2
+//
+// 1. > DUPLICATE_THRESHOLD of the nodes vote on some version of the the duplicate block 3,
+// but don't immediately duplicate confirm so they remove 3 from fork choice and reset PoH back to 1.
+// 2. All the votes on 3 don't land because there are no further blocks building off 3.
+// 3. Some < SWITCHING_THRESHOLD of nodes vote on 2, making it the heaviest fork because no votes on 3 landed
+// 4. Nodes then see duplicate confirmation on 3. 
+// 5. Unless somebody builds off of 3 to include the duplicate confirmed votes, 2 will still be the heaviest.
+// However, because 2 has < SWITCHING_THRESHOLD of the votes, people who voted on 3 can't switch, leading to a
+// stall 
+
+#[test]
+#[serial]
+#[allow(unused_attributes)]
+fn test_duplicate_shreds_switch_failure() {
+    solana_logger::setup_with_default(RUST_LOG_FILTER);
+    let validator_keypairs = vec![
+        "28bN3xyvrP4E8LwEgtLjhnkb7cY4amQb6DrYAbAYjgRV4GAGgkVM2K7wnxnAS7WDneuavza7x21MiafLu1HkwQt4",
+        "2saHBBoTkLMmttmPQP8KfBkcCw45S5cwtV3wTdGCscRC8uxdgvHxpHiWXKx4LvJjNJtnNcbSv5NdheokFFqnNDt8",
+        "4mx9yoFBeYasDKBGDWCTWGJdWuJCKbgqmuP8bN9umybCh5Jzngw7KQxe99Rf5uzfyzgba1i65rJW4Wqk7Ab5S8ye",
+        "2XFPyuzPuXMsPnkH98UNcQpfA7M4b2TUhRxcWEoWjy4M6ojQ7HGJSvotktEVbaq49Qxt16wUjdqvSJc6ecbFfZwj",
+    ]
+    .iter()
+    .map(|s| (Arc::new(Keypair::from_base58_string(s)), true))
+    .collect::<Vec<_>>();
+    let validators = validator_keypairs
+        .iter()
+        .map(|(kp, _)| kp.pubkey())
+        .collect::<Vec<_>>();
+
+    // Create 4 nodes:
+    // 1) Two nodes that sum to > DUPLICATE_THRESHOLD but < 2/3+ supermajority. It's important no
+    // one of them individually has > DUPLICATE_THRESHOLD to avoid duplicate confirming their own blocks
```

Comment:
you mean both of them *individually* have `< DUPLICATE_THRESHOLD` but the sum is `> DUPLICATE_THRESHOLD` right? Because it's important they duplicate confirm the block together so that we reset back to building on top of the duplicate block

Reply:
yep i misread the comment